app/assets/javascripts/covid19/src/webapp/components/proteome/SVGProteome.tsx

```diff
@@ -0,0 +1,244 @@
+import React from "react";
+import styled from "styled-components";
+import i18n from "../../../utils/i18n";
+import { Orf1aProts, Orf1bProts, RemainingGens } from "./PathGroups";
+import { ProteomePath } from "./ProteomePath";
+
+export interface VisibleGen {
+    orf1a?: boolean;
+    orf1b?: boolean;
+}
+
+interface SVGProteomeProps {
+    search: string;
+    setSearch: (value: string) => void;
+    visible: VisibleGen;
+    setVisible: (visible: VisibleGen) => void;
+    title: React.ReactNode;
+    setTitle: (value: React.ReactNode) => void;
+    setProteomeSelected: (value: boolean) => void;
+}
+
+export const SVGProteome: React.FC<SVGProteomeProps> = React.memo(props => {
+    const { search, setSearch, visible, setVisible, title, setTitle, setProteomeSelected } = props;
+    const [loading, setLoading] = React.useState(false);
+
+    const stateSetters = React.useMemo(
+        () => ({
+            setSearch,
+            setTitle,
+            setProteomeSelected,
+            setLoading,
+        }),
+        [setSearch, setTitle, setProteomeSelected]
+    );
+
+    const restoreVisible = React.useCallback(() => setVisible({}), [setVisible]);
+    const onMouseLeave = React.useCallback(() => {
+        setTitle(
+            <span>
+                {i18n.t("SARS-CoV-2")}
+                <br />
+                {i18n.t("Proteome")}
+            </span>
+        );
+        restoreVisible();
+    }, [setTitle, restoreVisible]);
+    const onClickOrf1a = React.useCallback(() => {
+        setVisible({ orf1a: true, orf1b: false });
+    }, [setVisible]);
```

Comment:
[best practices] [no need to change] the name of callback should reflect *what* they do, not the event that fired. Typically that's more declarative:

```
const setOrf1aVisible = useCallback(...)
...
return <Link onClick={setOrf1aVisible} />```
```